src/core/lis-modal-element.ts

```diff
@@ -0,0 +1,137 @@
+import {LitElement, html, css} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+import {unsafeHTML} from 'lit/directives/unsafe-html.js';
+
+
+/**
+ * @htmlElement `<lis-modal-element>`
+ *
+ * A Web Component that provides a generic modal element.
+ *
+ * @example
+ * The modal element's
+ * {@link name | `name`}, {@link heading | `heading`}, and
+ * {@link content | `content`} attributes/properties can be set via HTML or
+ * javascript.
```

Comment:
Also mention that the element is a wrapper for the UIkit modal and that you interact with the modal using the UIkit API (provide a link) and the modal-id property.